packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3649,7 +3648,7 @@ export class TransactionController extends BaseController<
       updatedTransactionParams =
         this.#checkIfTransactionParamsUpdated(transactionMeta);
 
-      if (skipHistory !== true) {
+      if (!this.isHistoryDisabled && skipHistory !== true) {
```

Comment:
Hi @OGPoyraz!

This conditional took me a second to grok, because:
a) expressions like !ishistoryDisabled are a double-negatives.
b) while `skipHistory !== true` is technically correct, it isn't typical javascript, though I understand that it was not part of this PR.

How about something like this:
```suggestion
    const shouldSkipHistory = this.isHistoryDisabled || skipHistory;

    if (!shouldSkipHistory) {
```

Reply:
That make sense Derek, thanks for suggestion.